lib/puppet/provider/yumrepo/inifile.rb

```diff
@@ -1,5 +1,56 @@
 require 'puppet/util/inifile'
 
+module Puppet::Provider::Yumrepo
+  module IniConfig
+    class Section < Puppet::Util::IniConfig::Section; end
+
+    class PhysicalFile < Puppet::Util::IniConfig::PhysicalFile
+      def store
+        unlinked = false
+        if @destroy_empty and (sections.empty? or sections.all?(&:destroy?))
+          ::File.unlink(@file)
+          unlinked = true
+        elsif sections.any?(&:dirty?)
+          text = self.format
+          @filetype.write(text)
+        end
+        sections.each(&:mark_clean)
+        return unlinked
+      end
+
+      private
+
+      def section_exists?(name)
+        if self.get_section(name)
+          true
+        elsif @file_collection and section = @file_collection.get_section(name) and !section.destroy?
+          true
+        else
+          false
+        end
+      end
+    end
+
+    class FileCollection < Puppet::Util::IniConfig::FileCollection
+      def store
+        @files.delete_if do |name, file|
+          file.store
+        end
+      end
+
+      private
+
+      # Create a new physical file and set required attributes on that file.
+      def new_physical_file(file)
+        @files[file] = PhysicalFile.new(file, destroy_empty: true)
+        @files[file].file_collection = self
+        @files[file]
+      end
+    end
+    File = FileCollection
```

Comment:
This looks to be copy-pasta'ed from https://github.com/puppetlabs/puppet/commit/622d46caf85a82941b80fc3b3785ff5aa93a1f9e#diff-7a57b4db77efed3a42f8c42cfa0d14243551e56d19f5622ed93821a02a30e4bdR304 That constant was necessary because we didn't want to break backwards compatibility. But since this is a new class, we can drop it.

Reply:
ah, I see it's used in `Puppet::Provider::Yumrepo::IniConfig::File.new`